### PR TITLE
feat(scenario-specification): activate specification candidate

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/scenario-specification/application/activate-candidate-specification.command.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Command } from '@nestjs-architects/typed-cqrs';
+import { Either } from 'fp-ts/Either';
+import {
+  NoCandidateToActivate,
+  SpecificationIsNoLongerACandidate,
+} from '../domain';
+
+export const ScenarioSpecificationNotFound = Symbol(
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/activate-candidate-specification.command.ts:8]: why symbols are started with a capital letter? I'd expect them to be a class or a type then, but It's a value

File: api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      const scenarioSpecification = await scenarioSpecificationRepo.find(
+        scenarioId,
+      );
+      expect(scenarioSpecification).toBeDefined();
+      expect(scenarioSpecification!.currentActiveSpecification).toEqual(
+        currentCandidateSpecificationId,
+      );
+      expect(
+        scenarioSpecification!.currentCandidateSpecification,
+      ).toBeUndefined();
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts:81]: it's checking more than only that it's saved
the name of the method is a bit misleading

File: api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    ThenSpecificationIsActivated: () => {
+      expect(events).toEqual([
+        new SpecificationActivated(currentCandidateSpecificationId),
+      ]);
+    },
+    ThenSpecificationIsNotActivated: async (
+      result: Either<ActivateError, void>,
+    ) => {
+      expect(result._tag).toEqual('Left');
```
Comment (Dyostiq) [inline api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts:91]: `expect(isLeft(result)).toBe(true)`

**Review by Dyostiq — APPROVED** [review]

File: api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts (hunk 3 above)
Comment (kgajowy) [reply api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/416/commits/4728b10a6bddf2263ed43523b720cc325d6ae2dc

File: api/apps/api/src/modules/scenario-specification/application/activate-candidate-specification.command.ts (hunk 1 above)
Comment (kgajowy) [reply api/apps/api/src/modules/scenario-specification/application/activate-candidate-specification.command.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/416/commits/4728b10a6bddf2263ed43523b720cc325d6ae2dc

File: api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/api/src/modules/scenario-specification/application/__tests__/activate-candidate-specification.fixtures.ts:81]: As we are in context of saving scenario specification after activating candidate, it verifies if the saved one changed indeed (i.e. persistence applied new version).

**Pull request merged**